src/main/java/com/zendesk/maxwell/MaxwellAbstractRowsEvent.java

```diff
@@ -205,13 +237,16 @@ public Object getData(String string) {
 	public List<RowMap> jsonMaps() {
 		ArrayList<RowMap> list = new ArrayList<>();
 		Object value;
-		for ( Row r : filteredRows()) {
+		for ( Iterator<Row> ri = filteredRows().iterator() ; ri.hasNext(); ) {
+			Row r = ri.next();
 			RowMap rowMap = new RowMap();
 
 			rowMap.setRowType(getType());
 			rowMap.setTable(getTable().getName());
 			rowMap.setDatabase(getDatabase().getName());
 			rowMap.setTimestamp(getHeader().getTimestamp() / 1000);
+			rowMap.setXid(getXid());
+			rowMap.setCommit(isCommit() && !ri.hasNext());
```

Comment:
That's what I was thinking, to strip the `boolean commit` param off of `setCommit`